packages/flutter/lib/src/widgets/animated_cross_fade.dart

```diff
@@ -189,49 +197,73 @@ class _AnimatedCrossFadeState extends State<AnimatedCrossFade> with TickerProvid
     }
   }
 
-  @override
-  Widget build(BuildContext context) {
-    List<Widget> children;
+  /// Whether we're in the middle of cross-fading right this frame.
+  bool get _isTransitioning => _controller.status == AnimationStatus.forward || _controller.status == AnimationStatus.reverse;
 
-    if (_controller.status == AnimationStatus.completed ||
-        _controller.status == AnimationStatus.forward) {
-      children = <Widget>[
-        new FadeTransition(
-          opacity: _secondAnimation,
-          child: widget.secondChild,
-        ),
-        new Positioned(
+  List<Widget> _buildCrossFadedChildren() {
+    const Key kFirstChildKey = const Key('first-child');
+    const Key kSecondChildKey = const Key('second-child');
+    final bool transitioningForwards = _controller.status == AnimationStatus.completed || _controller.status == AnimationStatus.forward;
+
+    Key topKey;
+    Widget topChild;
+    Animation<double> topAnimation;
+    Key bottomKey;
+    Widget bottomChild;
+    Animation<double> bottomAnimation;
+    if (transitioningForwards) {
+      topKey = kSecondChildKey;
+      topChild = widget.secondChild;
+      topAnimation = _secondAnimation;
+      bottomKey = kFirstChildKey;
+      bottomChild = widget.firstChild;
+      bottomAnimation = _firstAnimation;
+    } else {
+      topKey = kFirstChildKey;
+      topChild = widget.firstChild;
+      topAnimation = _firstAnimation;
+      bottomKey = kSecondChildKey;
+      bottomChild = widget.secondChild;
+      bottomAnimation = _secondAnimation;
+    }
+
+    return <Widget>[
+      new TickerMode(
+        key: bottomKey,
+        enabled: _isTransitioning,
+        child: new Positioned(
           // TODO(dragostis): Add a way to crop from top right for
           // right-to-left languages.
           left: 0.0,
           top: 0.0,
           right: 0.0,
-          child: new FadeTransition(
-            opacity: _firstAnimation,
-            child: widget.firstChild,
+          child: new ExcludeSemantics(
+            excluding: !_isTransitioning,
```

Comment:
you want to exclude the one that's going away and include the one that's coming in, regardless of the state of the animation.